src/core/toga/widgets/progressbar.py

```diff
@@ -17,16 +18,27 @@ def __init__(self, id=None, style=None, max=None, value=None, factory=None):
                 implementation of this class with the same name. (optional & normally not needed)
         """
         super().__init__(id=id, style=style, factory=factory)
+
+        self._value = None
         self._impl = self.factory.ProgressBar(interface=self)
 
         self.max = max
         self.value = value
-        self._running = False
+
         self.rehint()
 
+    @property
+    def running(self):
+        """
+        Returns:
+            True if the progress bar is running,
+            False if a specific value is displayed
+        """
+        return self.value is None
+
     @property
     def value(self):
-        """ The progress value
+        """ The progress value. Set to None to enter "running" mode.
```

Comment:
Apparently so? I think I misunderstood the behavior of a ProgressBar quite a bit. I expected ProgressBar to behave like this:

Two states:
 * "running mode", in which the progress is indeterminate (moving back and forth, spinning, orbiting, etc...). The `pulse()` method causes a GTK progress bar to move back and forth, which is why 50ms is used. In my proposed API, you trigger this mode by setting the interface's `value` to None. This made sense to me because None is an indeterminate value.
 * "value mode", when a specific value is set (ex: 50%). As far as I know, GTK performs no animation in this state. If any non-None value is assigned, the progress bar is put in this state.



Though I've never worked on Windows or Cocoa, I have definitely _seen_ progress bars animate with periodic flashes or similar, but It did not occur to me at the time of writing, and I did not realize that they could exist two different states while displaying a value. It sounds like I need to get the behavior definition straight before continuing.